datashader/reductions.py

```diff
@@ -139,26 +258,37 @@ def _finalize(bases, cuda=False, **kwargs):
         return xr.DataArray(bases[0], **kwargs)
 
 class by(Reduction):
-    """Apply the provided reduction separately per categorical ``column`` value.
+    """Apply the provided reduction separately per category.
     Parameters
     ----------
-    column : str
-        Name of the column to aggregate over. Column data type must be
-        categorical. Resulting aggregate has an outer dimension axis along the
-        categories present.
+    cats: str or CategoryPreprocess instance
+        Name of column to aggregate over, or a categorizer object that returns categories.
+        Resulting aggregate has an outer dimension axis along the categories present.
     reduction : Reduction
         Per-category reduction function.
     """
-    def __init__(self, cat_column, reduction):
-        self.columns = (cat_column, getattr(reduction, 'column', None))
+    def __init__(self, cats, reduction):
```

Comment:
Is it necessary to rename cat_column here? That seems to hurt backwards compatibility if anyone was supplying the keyword argument, and at least in typical cases the name is still accurate.

Reply:
Agreed, I don't remember why I even renamed it, but I put it back.